### PR TITLE
[CAT-102] byte stream support

### PR DESCRIPTION
**Pull request by goatrocks**
## Summary

This enables SDK users to manage their own file reading and upload to the backend by providing a map of filenames to  streams, so in addition to`files` on a `WorkflowSubmission`, you can add `io.BufferedIOBase` inheriting `streams` directly.

## Changelog (optional)
Modified WorkflowSubmission object to add optional streams
Modified Storage upload object to add optional streams
Modified client.py to handle the optional streams similar to files.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:
A quick checklist of things that should be done before the code is opened to review by others on the team.
Delete any that are not applicable.

- [x] Jira story/task is put into PR title
- [x] Code has been self-reviewed
- [x] Hard-to-understand areas have been commented
- [ ] Documentation has been updated
- [x] My changes generate no new warnings
- [x] Unit tests have been added for base functionality and known edge cases

### CODE REVIEW

File: indico/queries/storage.py [hunk 1]
```diff
@@ -124,5 +131,5 @@ def process_response(self, uploaded_files: List[dict]) -> List[str]:
         return urls
 
 
-# Alias to ensure backwards compatability
+# Alias to ensure backwards compatibility
```
Comment (fitzworkhub) [inline indico/queries/storage.py:134]: Compatability is how the brits write it ... like colour

File: indico/queries/workflow.py [hunk 2]
```diff
@@ -266,18 +270,22 @@ def __init__(
         submission: bool = True,
         bundle: bool = False,
         result_version: str = None,
+        streams: Dict[str, io.BufferedIOBase] = None
```
Comment (fitzworkhub) [inline indico/queries/workflow.py:273]: Can this keys for this dict be file paths?

**Review by fitzworkhub — APPROVED** [review]

File: indico/queries/workflow.py (hunk 2 above)
Comment (goatrocks) [reply indico/queries/workflow.py:273]: It could be if that's a requirement, and we could add some logic similar to the way we parse file names out for file paths, but ideally it would just be filename.ext
